src/main/java/com/google/sps/servlets/Util.java

```
@@ -0,0 +1,145 @@
+  private static final String API_KEY = getAPIKey(); 
+
+  public static List<ChromeOSDevice> getAllDevices(String userId) throws IOException {
+    final String refreshToken = getRefreshToken(userId);
+    File file = new File(Util.class.getResource(CLIENT_SECRET_FILE).getFile());
+    final GoogleClientSecrets clientSecrets =
+    GoogleClientSecrets.load(
+      JacksonFactory.getDefaultInstance(), new FileReader(file));
+    final String clientId = clientSecrets.getDetails().getClientId();
+    final String clientSecret = clientSecrets.getDetails().getClientSecret();
+    final String accessToken = getAccessToken(refreshToken, clientId, clientSecret);
+    ListDeviceResponse resp = getDevicesResponse(EMPTY_PAGE_TOKEN, accessToken);
+    final List<ChromeOSDevice> allDevices = new ArrayList<>(resp.getDevices());
+    while (resp.hasNextPageToken()) {
+      final String pageToken = (String) resp.getNextPageToken();
+      resp = getDevicesResponse(pageToken, accessToken);
+      allDevices.addAll(resp.getDevices());
+    }
+    return allDevices;
+  }
+
+  public static String getAPIKey() {
+    try {
+      File file = new File(Util.class.getResource(API_KEY_FILE).getFile());
+      String str = FileUtils.readFileToString(file);
+      return str;
+    } catch (IOException e) {
+      System.out.println(e.getMessage());
+    }
+    return EMPTY_API_KEY;
+  }
+
+  private static String getRefreshToken(String userId) throws IOException {
+    Query query = new Query("RefreshToken").setFilter(FilterOperator.EQUAL.of("userId", userId));
+    PreparedQuery results = datastore.prepare(query);
+    System.out.println(results.countEntities());
+    try {
+      Entity entity = results.asSingleEntity();
+      String refreshToken = (String) entity.getProperty("refreshToken");
+      return refreshToken;
+    } catch (PreparedQuery.TooManyResultsException e) {
+      System.out.println(e);
+      throw new IOException("Error while getting refresh token");
```

Let's throw TooManyResultsException directly instead of log and rethrow.